Solution1/Solution1.Tests/ConsoleApp/Service/UserCommunicationServiceTests.cs

```diff
@@ -0,0 +1,142 @@
+﻿using BusinessLayer.Abstract;
+using BusinessLayer.DTOs;
+using BusinessLayer.Service;
+using ConsoleApp.Service;
+using ConsoleApp.Service.Abstract;
+using Microsoft.Extensions.Logging;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Weather.Tests.ConsoleApp.Service
+{
+    public class UserCommunicationServiceTests
+    {
+        private readonly UserCommunicationService _userCommunicationService;
+        private readonly Mock<IWeatherServiсe> _weatherServiceMock;
+        private readonly Mock<ILogger> _loggerMock;
+
+        public UserCommunicationServiceTests()
+        {
+            _loggerMock = new Mock<ILogger>();
+            _weatherServiceMock = new Mock<IWeatherServiсe>();
+            _userCommunicationService = new UserCommunicationService(_loggerMock.Object, _weatherServiceMock.Object);
+        }
+
+        [Fact]
+        public void Communication_EnterEmptyCityName_ShowNoticeAndReturn()
+        {
+            // Arrange
+
+            //Act
+            var consoleOutput = new StringWriter();
```

Comment:
setup of console is arrange - all kinds of setup, from vars to mocks - are arranges